components/Ui/UiCta.vue

```diff
@@ -49,35 +48,43 @@ const props = withDefaults(defineProps<Props>(), {
 
 defineEmits(["click"]);
 
+const config = useRuntimeConfig();
+
 const urlString = computed(() => {
   if (typeof props.url === "string") {
     return props.url;
   }
+
   return props.url.url;
 });
 
-const iconPerLinkType = computed(() => {
-  const url = props.url;
+// Resolving link type (Based on https://github.com/nuxt/nuxt/blob/v3.6.3/packages/nuxt/src/app/components/nuxt-link.ts#L179)
+const isExternalLogically = computed<boolean>(() => {
+  return urlString.value === "" || urlHasProtocol(urlString.value);
+});
 
-  if (props.label === "Under construction") {
-    return ErrorOutline16;
+const isExternalVisually = computed<boolean>(() => {
+  // Same domain
+  if (urlString.value.startsWith(config.public.siteUrl)) {
+    return false;
   }
-  if (isExternal(url as string)) {
+
+  return isExternalLogically.value;
+});
+
+const isFragment = computed<boolean>(() => {
```

Comment:
just curious if there's a difference between fragment and anchor

Reply:
As I understand it, _anchor_ is the `<a>` element and what we're referring to here is if the anchor points to a _document fragment_ via the `href` attribute (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a?retiredLocale=de#href).